sdk/storage/azblob/blob/models.go

```diff
@@ -258,6 +258,32 @@ func (o *UndeleteOptions) format() *generated.BlobClientUndeleteOptions {
 
 // ---------------------------------------------------------------------------------------------------------------------
 
+type BlobDeleteType = generated.DeleteType
+
+type PermanentDeleteOptions struct {
+	DeleteOptions
+	BlobDeleteType *BlobDeleteType
```

Comment:
DeleteType only has two valid values "none" or "permanent". As the api name itself is saying permanent delete, this option is redundant. We can skip taking this option here.